### PR TITLE
[OP] Add Mathematical functions

### PR DESCRIPTION
**Pull request by yajiedesign**
add 
- hypot 
- degrees 
- radians 
- cosh 
- sinh 
- tanh 
- arcsinh 
- arccosh 
- arctanh

https://github.com/dmlc/mxnet/issues/3201

### CODE REVIEW

File: src/operator/mshadow_op.h [hunk 1]
```diff
@@ -12,6 +12,8 @@
 namespace mxnet {
 namespace op {
 namespace mshadow_op {
+#define PI 3.14159265358979323846
```
Comment (piiswrong) [inline src/operator/mshadow_op.h]: const float PI = ....;
macros contaminates global namespace

File: src/operator/mshadow_op.h [hunk 2]
```diff
@@ -12,6 +12,13 @@
 namespace mxnet {
 namespace op {
 namespace mshadow_op {
+#ifdef __CUDA_ARCH__
+#define CONSTANT __constant__
+#else
+#define CONSTANT
```
Comment (yajiedesign) [inline src/operator/mshadow_op.h]: no constant can't build on cuda. constexpr vs2013 not support.
i try change it to
#ifdef __CUDA_ARCH__
__constant__ const float PI = 3.14159265358979323846;
#else
const float PI = 3.14159265358979323846;
#endif

Comment (piiswrong) [inline src/operator/mshadow_op.h]: __constant__ makes it a actual variable, which is bad for performance.
I guess let's use #define MXNET_PI then.

Comment (yajiedesign) [inline src/operator/mshadow_op.h]: I am show the assembly code in cuda .in release it not actual variable.

Comment (piiswrong) [inline src/operator/mshadow_op.h]: ok

**Pull request merged**